src/axom/slic/core/Logger.cpp

```diff
@@ -183,31 +178,35 @@ void Logger::addStreamToAllMsgLevels(LogStream* ls)
 
   for(int level = message::Error; level < message::Num_Levels; ++level)
   {
-    this->addStreamToMsgLevel(ls, static_cast<message::Level>(level));
-
-  }  // END for all levels
+    this->addStreamToMsgLevel(ls,
+                              static_cast<message::Level>(level),
+                              pass_ownership);
+  }
 }
 
 //------------------------------------------------------------------------------
-void Logger::addStreamToAllTags(LogStream* ls)
+void Logger::addStreamToAllTags(LogStream* ls, bool pass_ownership)
 {
   if(ls == nullptr)
   {
     std::cerr << "WARNING: supplied log stream is NULL!\n";
     return;
   }
 
-  if(ls == nullptr)
+  if(m_taggedStreams.empty())
   {
     std::cerr << "WARNING: no tags are available!\n";
+    if(pass_ownership)
+    {
+      m_streamObjectsManager[ls] = ls;
+    }
+
     return;
   }
```

Comment:
@bmhan12 -- this fixes a potential leak in the new `slic::addStreamToAllTags` function.

Since we cannot add tags to `slic` without attaching a `LogStream` (via the `slic::addStreamToTag()` function), it's not immediately clear to me how this function is intended to be used. Perhaps a clearer example demonstrating its intended usage would help.